metricflow/sql/sql_exprs.py

```diff
@@ -1144,3 +1149,70 @@ def matches(self, other: SqlExpressionNode) -> bool:  # noqa: D
         if not isinstance(other, SqlRatioComputationExpression):
             return False
         return self._parents_match(other)
+
+
+class SqlBetweenExpression(SqlExpressionNode):
+    """A BETWEEN clause like `column BETWEEN val1 AND val2`"""
+
+    def __init__(  # noqa: D
+        self, column_arg: SqlExpressionNode, left_value: SqlExpressionNode, right_value: SqlExpressionNode
+    ) -> None:
+        self._column_arg = column_arg
+        self._left_value = left_value
+        self._right_value = right_value
+        super().__init__(node_id=self.create_unique_id(), parent_nodes=[column_arg, left_value, right_value])
+
+    @classmethod
+    def id_prefix(cls) -> str:  # noqa: D
+        return SQL_EXPR_BETWEEN_PREFIX
+
+    @property
+    def requires_parenthesis(self) -> bool:  # noqa: D
+        return False
+
+    def accept(self, visitor: SqlExpressionNodeVisitor[VisitorOutputT]) -> VisitorOutputT:  # noqa: D
+        return visitor.visit_between_expr(self)
+
+    @property
+    def description(self) -> str:  # noqa: D
+        return "BETWEEN operator"
+
+    @property
+    def column_arg(self) -> SqlExpressionNode:  # noqa: D
+        return self._column_arg
+
+    @property
+    def left_value(self) -> SqlExpressionNode:  # noqa: D
+        assert isinstance(
```

Comment:
Actually, other expressions should be possible. e.g. `WHERE ds BETWEEN some_date_column AND some_other_date_column`

Reply:
removed this assert!